### PR TITLE
#11610: Add new sweep test workflow to CI

### LINKED ISSUE #11610: Sweeps: Add CI workflow for executing all sweep tests
- Basic first workflow to execute all existing and current sweep test vectors on GitHub runners via a manual dispatch

### PR DESCRIPTION
**Pull request by jdesousa-TT**
### Ticket
#11610 

### What's changed
- Added a CI workflow to execute the new sweep framework on CI runners
- Add some changes to how tt-smi is called in the runner to allow it to work on both lab machines and CI runners

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by tt-rkim** [comment]
Pls overwrite the current ttnn sweeps file and let me @eyonland know
I hate `<anything>-new`

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (ttmchiou) [inline .github/workflows/ttnn-run-sweeps-new.yaml:10]: is this meant to target this branch on every push?

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (ttmchiou) [inline .github/workflows/ttnn-run-sweeps-new.yaml:30]: @tt-rkim is this environment: dev still in use?

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (ttmchiou) [inline .github/workflows/ttnn-run-sweeps-new.yaml:40]: we have a prepare-metal-run github action which simplifies some of these steps in .github/actions

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (ttmchiou) [inline .github/workflows/ttnn-run-sweeps-new.yaml:44]: please adjust timeout-minutes to a proper length after all CI tests pass with some buffer time.
Typically an extra 50% of current total time to completion is a decent rule of thumb

File: .github/workflows/ttnn-run-sweeps.yaml
Comment (tt-rkim) [inline .github/workflows/ttnn-run-sweeps.yaml:45]: 720?

Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps.yaml:45]: 12 hours, we will need more than this probably too. There is a lot of tests, at this point just the matmuls and a couple eltwise ops take 4.5 hours to run.

Comment (tt-rkim) [reply .github/workflows/ttnn-run-sweeps.yaml:45]: That's tough - didn't know it was that much

Please put `timeout-minutes` at the job level, rather than the step level. If you put it at the step level as you do here, then it will be ignored. Any number you put for `steps.<step>.timeout-minutes:` that's higher than 360 will be ignored.

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps-new.yaml]: Just temporary to test the workflow, will remove before merging.

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps-new.yaml]: Changed to use that action.

File: .github/workflows/ttnn-run-sweeps.yaml
Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps.yaml]: Moved to job level.

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (tt-rkim) [reply .github/workflows/ttnn-run-sweeps-new.yaml]: No, please rid of it

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps-new.yaml]: Addressed in Raymond's comment below.

**Comment by tt-rkim** [comment]
Wanna schedule a run for late tonight so we can see what happened in the morning? Then we'll approve tomorrow morning once everything is all gucci?

File: tests/sweep_framework/tt_smi_util.py
Comment (TT-billteng) [inline tests/sweep_framework/tt_smi_util.py:1]: @vtangTT can you re-use this code for your pytest hook?

File: .github/workflows/ttnn-run-sweeps-new.yaml
Comment (jdesousa-TT) [reply .github/workflows/ttnn-run-sweeps-new.yaml]: Is there a diff environment? Or just remove it completely?

**Comment by jdesousa-TT** [comment]
> Wanna schedule a run for late tonight so we can see what happened in the morning? Then we'll approve tomorrow morning once everything is all gucci?

@tt-rkim I did test-run it here https://github.com/tenstorrent/tt-metal/actions/runs/10475412797 and just cancelled once it started with all the sweeps to not take up two runners for 4 hours. If you want to do a full run I can tonight but as far as the workflow itself goes I think everything is fine.

**Review by tt-rkim — APPROVED** [review]

**Review by sjameelTT — APPROVED** [review]
Thanks for the cloud machine fix

File: tests/sweep_framework/tt_smi_util.py
Comment (vtangTT) [reply tests/sweep_framework/tt_smi_util.py:1]: Yes, it's pretty similar to the one I implemented in `conftest.py`. I don't think this implementation provides a way to reset individual devices though

Comment (TT-billteng) [reply tests/sweep_framework/tt_smi_util.py:1]: @jdesousa-TT and @vtangTT can you guys try to merge this code in a diff PR?

**Pull request merged**
